File: services/payment/Payment.Domain/Payments/Payment.cs
namespace Payment.Domain.Payments;

public sealed class Payment
{
    public Guid Id { get; set; }

    public Guid OrderId { get; set; }

    public long AmountMinor { get; set; }

    public string Currency { get; set; } = string.Empty;

    public PaymentStatus Status { get; set; }

    public string Provider { get; set; } = string.Empty;

    public string? ProviderReference { get; set; }

    public string? FailureReason { get; set; }

    public DateTimeOffset CreatedAt { get; set; }

    public DateTimeOffset UpdatedAt { get; set; }

    public DateTimeOffset? SucceededAt { get; set; }

    public DateTimeOffset? FailedAt { get; set; }

    public DateTimeOffset? CancelledAt { get; set; }
}

File: frontend/angular-app/src/app/features/payments/models/payment.ts
export type PaymentDetails = {
  id: string;
  orderId: string;
  amountMinor: number;
  currency: string;
  status: string;
  provider: string;
  providerReference: string | null;
  failureReason: string | null;
  createdAt: string;
  updatedAt: string;
  succeededAt: string | null;
  failedAt: string | null;
  cancelledAt: string | null;
};

export type CreatePaymentRequest = {
  orderId: string;
  amountMinor: number;
  currency: string;
  provider: string;
};

export type SucceedPaymentRequest = {
  providerReference: string;
};
